File: DONGUK/프로그래머스_WEEK_11/N개의 최소공배수.py
def solution(arr):
    answer = 0
    
    num = max(arr)
    while True:
        flag = True
        for i in range(len(arr)):
            if num % arr[i] != 0:
                flag = False
                break
        
        if flag:
            answer = num
            break
        
        num += 1
        
    return answer

File: JUNHEE/프로그래머스_week11/N개의 최소공배수.cpp
#include <string>
#include <vector>

using namespace std;
int GCD(int a, int b){
    int temp;
    if(a < b){
        temp = a;
        a = b;
        b = temp;
    }
    temp = 0;
    while(b != 0){
        temp = a % b;
        a = b;
        b = temp;
    }
    return a;
}
int solution(vector<int> arr) {
    int answer = arr[0];
    for(int i = 0; i < arr.size(); i++){
        int t_gcd = GCD(answer, arr[i]);
        int t_lcm = answer * arr[i] / t_gcd;
        answer = t_lcm;
    }
    return answer;
}
